src/test/java/gregtech/api/capability/impl/AbstractRecipeLogicTest.java

```diff
@@ -136,4 +145,38 @@ protected long getMaxVoltage() {
         MatcherAssert.assertThat(AbstractRecipeLogic.areItemStacksEqual(arl.getOutputInventory().getStackInSlot(0),
                 new ItemStack(Blocks.STONE, 1)), is(true));
     }
+
+    @Test
+    public void trySearchNewRecipe_testEdgeCase() {
+        RecipeMaps.CHEMICAL_RECIPES.recipeBuilder()
+                .fluidInputs(Materials.SulfurDioxide.getFluid(1000), Materials.Oxygen.getFluid(1000))
+                .notConsumable(OreDictUnifier.get(OrePrefix.dust, Materials.Iron))
```

Comment:
OreDictUnifier.get(OrePrefix.dust, Iron) returns a empty itemstack in the test, which adds an empty stack as key.
During lookup empty itemstacks are skipped.

Changing to another itemstack here and when adding to the  itemList.add() below makes the test pass